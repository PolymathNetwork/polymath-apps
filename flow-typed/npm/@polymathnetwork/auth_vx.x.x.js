// flow-typed signature: d731cb9b18720f4cf204447fa5d4d547
// flow-typed version: <<STUB>>/@polymathnetwork/auth_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/auth'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/auth' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/auth/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/actions' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/helpers' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/networks' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/PolymathAuth' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/reducer' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/auth/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/babel.config'>;
}
declare module '@polymathnetwork/auth/src/actions.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/actions'>;
}
declare module '@polymathnetwork/auth/src/helpers.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/helpers'>;
}
declare module '@polymathnetwork/auth/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/index'>;
}
declare module '@polymathnetwork/auth/src/networks.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/networks'>;
}
declare module '@polymathnetwork/auth/src/PolymathAuth.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/PolymathAuth'>;
}
declare module '@polymathnetwork/auth/src/reducer.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/reducer'>;
}
declare module '@polymathnetwork/auth/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/webpack.dev'>;
}
declare module '@polymathnetwork/auth/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/webpack.prod'>;
}
