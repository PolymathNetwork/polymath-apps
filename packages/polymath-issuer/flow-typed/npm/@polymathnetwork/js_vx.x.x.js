// flow-typed signature: f74591e4e08edd90ce9c87e65e769c1d
// flow-typed version: <<STUB>>/@polymathnetwork/js_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/js'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/js' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/js/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/Contract' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/IModuleFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PermissionManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PolyToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/STO' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TickerRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/IPFS' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/types' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/js/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/babel.config'>;
}
declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CappedSTOFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/Contract.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/Contract'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/IModuleFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/IModuleFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PermissionManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PermissionManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PolyToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PolyToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityTokenRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/STO.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/STO'>;
}
declare module '@polymathnetwork/js/src/contracts/TickerRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TickerRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/TransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TransferManager'>;
}
declare module '@polymathnetwork/js/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/index'>;
}
declare module '@polymathnetwork/js/src/IPFS.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/IPFS'>;
}
declare module '@polymathnetwork/js/src/types.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/types'>;
}
declare module '@polymathnetwork/js/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.dev'>;
}
declare module '@polymathnetwork/js/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.prod'>;
}
