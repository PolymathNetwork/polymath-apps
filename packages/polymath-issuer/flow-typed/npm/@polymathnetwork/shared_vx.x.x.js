// flow-typed signature: 1b60d8488df603980ab0f1307a4a306d
// flow-typed version: <<STUB>>/@polymathnetwork/shared_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/shared'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/shared' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x'
  >;
}
