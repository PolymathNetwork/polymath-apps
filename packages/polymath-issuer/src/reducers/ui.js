// @flow
import { LOCATION_CHANGE } from 'connected-react-router';

export type UIState = {
  header: {
    variant: 'default' | 'transparent',
  },
  footer: {
    variant: 'default' | 'transparent',
  },
};

const defaultState: UIState = {
  header: {
    variant: 'default',
  },
  footer: {
    variant: 'default',
  },
};

export default (state: UIState = defaultState, action) => {
  if (action.type === LOCATION_CHANGE) {
    const pathname = action.payload.location.pathname;

    if (pathname === '/') {
      return {
        ...state,
        header: {
          variant: 'transparent',
        },
        footer: {
          variant: 'transparent',
        },
      };
    }

    return defaultState;
  } else {
    return state;
  }
};
