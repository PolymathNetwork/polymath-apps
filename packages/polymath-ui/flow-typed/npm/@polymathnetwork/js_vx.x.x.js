// flow-typed signature: ebaa6de5186b7780234aa39ddaeadc86
// flow-typed version: <<STUB>>/@polymathnetwork/js_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/js'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/js' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/js/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/cli_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/node_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-export-default-from_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-object-rest-spread_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-transform-react-jsx_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/preset-env_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/preset-flow_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@babel/runtime-corejs2_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/@polymathnetwork/shared_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/babel-core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/bignumber.js_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/bs58_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/flow-bin_v0.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/husky_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/jest_v22.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/polymath-core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/redux_v3.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/redux-thunk_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/web3_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/webpack_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/flow-typed/npm/webpack-cli_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/Contract' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/IModuleFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PermissionManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PolyToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/STO' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TickerRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/IPFS' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/types' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/js/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/babel.config'>;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/cli_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/cli_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/core_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/node_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/node_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-export-default-from_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-export-default-from_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-object-rest-spread_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/plugin-proposal-object-rest-spread_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/plugin-transform-react-jsx_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/plugin-transform-react-jsx_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/preset-env_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/preset-env_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/preset-flow_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/preset-flow_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@babel/runtime-corejs2_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@babel/runtime-corejs2_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/@polymathnetwork/shared_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/@polymathnetwork/shared_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/babel-core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/babel-core_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/bignumber.js_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/bignumber.js_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/bs58_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/bs58_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/flow-bin_v0.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/flow-bin_v0.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/husky_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/husky_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/jest_v22.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/jest_v22.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/polymath-core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/polymath-core_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/redux_v3.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/redux_v3.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/redux-thunk_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/redux-thunk_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/web3_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/web3_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/webpack_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/webpack_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/flow-typed/npm/webpack-cli_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/flow-typed/npm/webpack-cli_vx.x.x'
  >;
}
declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/CappedSTOFactory'
  >;
}
declare module '@polymathnetwork/js/src/contracts/Contract.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/Contract'
  >;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManager.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/CountTransferManager'
  >;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/CountTransferManagerFactory'
  >;
}
declare module '@polymathnetwork/js/src/contracts/IModuleFactory.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/IModuleFactory'
  >;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/PercentageTransferManager'
  >;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory'
  >;
}
declare module '@polymathnetwork/js/src/contracts/PermissionManager.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/PermissionManager'
  >;
}
declare module '@polymathnetwork/js/src/contracts/PolyToken.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/PolyToken'
  >;
}
declare module '@polymathnetwork/js/src/contracts/SecurityToken.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/SecurityToken'
  >;
}
declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/SecurityTokenRegistry'
  >;
}
declare module '@polymathnetwork/js/src/contracts/STO.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/STO'>;
}
declare module '@polymathnetwork/js/src/contracts/TickerRegistry.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/TickerRegistry'
  >;
}
declare module '@polymathnetwork/js/src/contracts/TransferManager.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/js/src/contracts/TransferManager'
  >;
}
declare module '@polymathnetwork/js/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/index'>;
}
declare module '@polymathnetwork/js/src/IPFS.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/IPFS'>;
}
declare module '@polymathnetwork/js/src/types.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/types'>;
}
declare module '@polymathnetwork/js/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.dev'>;
}
declare module '@polymathnetwork/js/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.prod'>;
}
