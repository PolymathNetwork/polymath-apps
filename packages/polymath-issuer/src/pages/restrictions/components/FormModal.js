// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import { Form } from 'carbon-components-react';
import { Remark, Modal, Button } from '@polymathnetwork/ui';
import validator from '@polymathnetwork/ui/validator';
import { addAddressToTransferManager } from '../../../actions/compliance';
import GlobalRestrictionsForm from './GlobalRestrictionsForm';

type Props = {
  isOpen: boolean,
  handleClose: () => any,
  restrictionType: string,
};

export class FormModal extends Component<Props> {
  render() {
    const { isOpen, handleClose, restrictionType } = this.props;
    return (
      <Modal isOpen={isOpen} onClose={handleClose}>
        <Modal.Body>
          {restrictionType === 'custom' && (
            <h1 className="pui-h2">
              Configure Custom Trade Volume Restriction
            </h1>
          )}
          {restrictionType === '24h' && (
            <h1 className="pui-h2">Configure Daily Trade Volume Restriction</h1>
          )}
          <h3 className="pui-h3">
            The volume restriction can be specified in number of tokens or as a
            percentage of total supply and will be enforced only between the
            Start date and time and the End date and time.
          </h3>
          <Remark title="Note">
            — Volume restrictions do not apply to primary issuance, only to
            secondary trades.
            <br />— The Global Restriction applies to each Investor
            individually, not the aggregate of all sales across all investors
            within the period.
          </Remark>
          <GlobalRestrictionsForm
            handleClose={handleClose}
            restrictionType={restrictionType}
          />
        </Modal.Body>
      </Modal>
    );
  }
}

export default FormModal;
