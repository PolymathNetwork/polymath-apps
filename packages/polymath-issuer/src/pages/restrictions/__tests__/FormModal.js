import FormModal from '../components/FormModal';
import renderer from 'react-test-renderer';
import React from 'react';
import configureStore from 'redux-mock-store';
import { theme, GlobalStyles } from '@polymathnetwork/ui';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import moment from 'moment';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);

describe('Individual Restrictions Component', () => {
  it('should render 24h title', () => {
    const props = {
      restrictionType: '24h',
    };
    const tree = shallow(<FormModal {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render custom title', () => {
    const props = {
      restrictionType: 'custom',
    };
    const tree = shallow(<FormModal {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
