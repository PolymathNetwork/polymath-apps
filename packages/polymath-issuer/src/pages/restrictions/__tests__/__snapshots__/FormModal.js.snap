// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Individual Restrictions Component should render 24h title 1`] = `
<Modal__StyledModal>
  <Body>
    <h1
      className="pui-h2"
    >
      Configure Daily Trade Volume Restriction
    </h1>
    <h3
      className="pui-h3"
    >
      The volume restriction can be specified in number of tokens or as a percentage of total supply and will be enforced only between the Start date and time and the End date and time.
    </h3>
    <Remark
      title="Note"
    >
      — Volume restrictions do not apply to primary issuance, only to secondary trades.
      <br />
      — The Global Restriction applies to each Investor individually, not the aggregate of all sales across all investors within the period.
    </Remark>
    <Connect(WithFormik(AddGlobalRestrictionsComponent))
      restrictionType="24h"
    />
  </Body>
</Modal__StyledModal>
`;

exports[`Individual Restrictions Component should render custom title 1`] = `
<Modal__StyledModal>
  <Body>
    <h1
      className="pui-h2"
    >
      Configure Custom Trade Volume Restriction
    </h1>
    <h3
      className="pui-h3"
    >
      The volume restriction can be specified in number of tokens or as a percentage of total supply and will be enforced only between the Start date and time and the End date and time.
    </h3>
    <Remark
      title="Note"
    >
      — Volume restrictions do not apply to primary issuance, only to secondary trades.
      <br />
      — The Global Restriction applies to each Investor individually, not the aggregate of all sales across all investors within the period.
    </Remark>
    <Connect(WithFormik(AddGlobalRestrictionsComponent))
      restrictionType="custom"
    />
  </Body>
</Modal__StyledModal>
`;
